tests/test_estimate_params.py

```diff
@@ -0,0 +1,648 @@
+# National Aeronautics and Space Administration.  All Rights Reserved.
+
+from copy import deepcopy
+import io
+import numpy as np
+from os.path import dirname, join
+import pickle
+import sys
+import unittest
+
+# This ensures that the directory containing ProgModelTemplate is in the python search directory
+sys.path.append(join(dirname(__file__), ".."))
+
+from prog_models import *
+from prog_models.models import *
+from prog_models.models.test_models.linear_models import (
+    OneInputNoOutputNoEventLM, OneInputOneOutputNoEventLM, OneInputNoOutputOneEventLM, OneInputOneOutputNoEventLMPM)
+from prog_models.models.thrown_object import LinearThrownObject
+
+class TestEstimateParams(unittest.TestCase):
+    def test_estimate_params_works(self):
+        m = ThrownObject()
+        results = m.simulate_to_threshold(save_freq=0.5)
+        data = [(results.times, results.inputs, results.outputs)]
+        gt = m.parameters.copy()
+
+        self.assertTrue(m.parameters == gt)
+        self.assertEqual(m.parameters, gt)
+
+        # Now lets reset some parameters
+        m.parameters['thrower_height'] = 1.5
+        m.parameters['throwing_speed'] = 25
+        keys = ['thrower_height', 'throwing_speed']
+        m.estimate_params(data, keys)
+        for key in keys: # using assert not equal also works.
+            self.assertAlmostEqual(m.parameters[key], gt[key], 2)
+        
+        m.estimate_params(data, keys, bounds=((0, 4), (20, 42)))
+        # Enforce if parameters are within bounds after estimate_params()
+        check = m.calc_error(results.times, results.inputs, results.outputs)
+
+        # Demonstrates furhter limitations of Parameter Estiamtion
+
+        before = m.parameters
+        m.estimate_params(data, keys, bounds=((1.243, 4.0), (-12.9234, 3.33333333)))
+        check3 = m.calc_error(results.times, results.inputs, results.outputs)
+        after = m.parameters
+
+        # Now with limits that do include the true values
+        # only returning the upper bounds? Why after this defined behavior?
+        # m.parameters['throwing_speed'] = 15
+
+        # Or two calls to estimate_params would resolve this issue
+        m.estimate_params(data, keys, bounds=((0, 8), (20, 42)))
+        # m.estimate_params(data, keys, bounds=(0, 8), (20, 42))
+        ax = m.parameters
+        value3 = m.calc_error(results.times, results.inputs, results.outputs)
+        for key in keys:
+            self.assertAlmostEqual(m.parameters[key], gt[key], 2, "Limits of the Bounds do not include the true values")
+        
+        # self.assertTrue(m.parameters == gt)
+
+
+    def test_estimate_params(self):
+        m = ThrownObject()
+        results = m.simulate_to_threshold(save_freq=0.5)
+        data = [(results.times, results.inputs, results.outputs)]
+        gt = m.parameters.copy()
+
+        # Reset some parameters
+        m.parameters['thrower_height'] = 1.5
```

Comment:
I think lines 72 - 78 are the same as in test_estimate_params_works, is that right? If so, we only need to check it once